packages/NodeNameResolver/NodeNameResolver/ClassNameResolver.php

```diff
@@ -33,7 +39,10 @@ public function getNode(): string
      */
     public function resolve(Node $node): ?string
     {
-        if (property_exists($node, 'namespacedName')) {
+        if (in_array($node::class, self::ALLOWED_CLASSLIKE, true) && property_exists(
+            $node,
+            'namespacedName'
+        ) && $node->namespacedName instanceof Name) {
```

Comment:
I've just noticed the `ClassLike` is on the input, so it's always ClassLike here:

```suggestion
        if ($node->namespacedName instanceof Name) {
```

Reply:
updated, property_exists still needed or it will cause test error https://github.com/rectorphp/rector-src/runs/4378240342?check_suite_focus=true#step:11:18